File: src/floww/core/workspace/manager.py
from .backends import WorkspaceBackend
from .factory import create_backend


class WorkspaceManager:
    """Delegates workspace operations to a compositor-specific backend."""

    def __init__(self, backend: WorkspaceBackend | None = None):
        self._backend = backend if backend is not None else create_backend()

    def switch(self, target: int) -> bool:
        """Switch to workspace ``target``. Returns True on success, False otherwise."""
        return self._backend.switch(int(target))

    def get_total_workspaces(self) -> int:
        return self._backend.get_total_workspaces()

    def get_append_base_offset(self) -> int:
        return self._backend.get_append_base_offset()

File: internal/workspace/manager.go
package workspace

import "github.com/dagimg-dot/floww/internal/workspace/backends"

// WorkspaceManager delegates workspace operations to a compositor-specific backend.
type WorkspaceManager struct {
	backend backends.WorkspaceBackend
}

// NewWorkspaceManager creates a WorkspaceManager with the given backend.
func NewWorkspaceManager(backend backends.WorkspaceBackend) *WorkspaceManager {
	return &WorkspaceManager{backend: backend}
}

// Switch to workspace target. Returns true on success.
func (m *WorkspaceManager) Switch(target int) bool {
	return m.backend.Switch(target)
}

// GetTotalWorkspaces returns the backend-specific notion of workspace count.
func (m *WorkspaceManager) GetTotalWorkspaces() int {
	return m.backend.GetTotalWorkspaces()
}

// GetAppendBaseOffset returns the offset added to workflow targets when append is true.
func (m *WorkspaceManager) GetAppendBaseOffset() int {
	return m.backend.GetAppendBaseOffset()
}
